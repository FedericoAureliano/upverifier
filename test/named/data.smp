print(e.source)
print(f(e))

data mRef := A | B | C
data eKind := eRequest | eResponse | eCommand

function e : Event := cons(A, C, eCommand, cons(B, A, eResponse, cons(A, B, eRequest, nil)))

data Event := 
  | cons {source: mRef, target: mRef, kind: eKind, history: Event} 
  | nil 

function f(e: Event): Int :=
  if e.source = A then 
    0 
  else 
    1

recursive function length1 (e: Event) : Int := 
  if (e is cons) then
    1 + length1(e.history)
  else
    0

recursive function length2 (e: Event) : Int := 
  if (e is nil) then
    0
  else
    1 + length2(e.history)

print(e)
print(length1(e))
print(length1(e) = length2(e))