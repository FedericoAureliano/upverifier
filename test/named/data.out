A
0
cons(A, C, eCommand, cons(B, A, eResponse, cons(A, B, eRequest, nil)))
3
true
